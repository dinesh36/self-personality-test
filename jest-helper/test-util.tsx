import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import { render } from '@testing-library/react';
import { getStore } from '../src/store/store/store';

export const getComponentWithReduxWrapper = (cmp: React.ReactNode) => {
  const store = getStore();
  return renderer.create(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};

export const getComponentWithReduxWrapperForReactTestingLib = (cmp: React.ReactNode) => {
  const store = getStore();
  return render(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};
