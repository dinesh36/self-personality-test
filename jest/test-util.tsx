import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import React from 'react';
import { getStore } from '../src/store/store/store';

export const getComponentWithReduxWrapper = (cmp: React.ReactNode) => {
  const store = getStore();
  return renderer.create(
    <Provider store={store}>
      {cmp}
    </Provider>,
  );
};
