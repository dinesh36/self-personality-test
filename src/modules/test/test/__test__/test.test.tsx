import { getComponentWithReduxWrapper } from '@jestHelper/test-util';
import Test from '../test';

describe('Test Component', () => {
  it('Should render the test component', () => {
    const component = getComponentWithReduxWrapper(<Test />);
    const cmp = component.toJSON();

    expect(cmp).toMatchSnapshot();
  });
});
