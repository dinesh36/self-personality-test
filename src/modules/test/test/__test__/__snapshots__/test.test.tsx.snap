// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component Should render the test component 1`] = `
<div
  className="test"
>
  <div
    className="loadingContainer"
  >
    <div
      aria-busy={true}
      aria-live="polite"
      className="ant-spin ant-spin-lg ant-spin-spinning css-dev-only-do-not-override-1n7nwfa"
    >
      <span
        className="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
